feat: add entitlementV2 component

Ali-D-Akbar opened the pull request:
[PROD-2463](https://openedx.atlassian.net/browse/PROD-2463)

This PR adds a new design for Entitlement Component while keeping the older components. This v2 component uses the same Table component that we used in Enrollment v2 component.

Entitlement with Expanded Row:
<img width="1626" alt="Screenshot 2021-09-09 at 11 32 36 PM" src="https://user-images.githubusercontent.com/52413434/132742993-0af16a5a-cbb9-4f93-8c7f-0bd445c92c5a.png">

Entitlement with Disabled Reissue and Enabled Expire actions:
<img width="1646" alt="Screenshot 2021-09-09 at 11 33 06 PM" src="https://user-images.githubusercontent.com/52413434/132743081-32aa0f2a-f545-47a9-a1eb-5ea007e50323.png">

DawoudSheraz commented on src/users/entitlements/v2/Entitlements.jsx:
```diff
@@ -0,0 +1,298 @@
+          {entitlement.courseUuid}
+        </Hyperlink>
+      ),
+      mode: entitlement.mode,
+      enrollment: (entitlement.enrollmentCourseRun ? (
+        <a
+          href={`${getConfig().LMS_BASE_URL}/courses/${entitlement.enrollmentCourseRun}`}
+          rel="noopener noreferrer"
+          target="_blank"
+        >
+          {entitlement.enrollmentCourseRun}
+        </a>
+      ) : 'Course Run Not Selected'),
+      expiredAt: formatDate(entitlement.expiredAt),
+      created: formatDate(entitlement.created),
+      modified: formatDate(entitlement.modified),
```
keep the sequence and column titles the same as the design.

DawoudSheraz commented on src/users/entitlements/v2/Entitlements.jsx:
```diff
@@ -0,0 +1,298 @@
+                setUserEntitlement(entitlement);
+                setFormType(REISSUE);
+              }}
+              disabled={Boolean(!entitlement.enrollmentCourseRun)}
```
Does disable work with item? Instead of disabling it, we can skip the item altogether if the action is not applicable.

Ali-D-Akbar replied:
This is how it works. Reissue is disabled and expired is enabled. I think it still makes sense to keep the action there just so the support user knows how many actions are possible on each entitlement.

<img width="193" alt="Screenshot 2021-09-09 at 11 25 44 PM" src="https://user-images.githubusercontent.com/52413434/132741999-86be7d8b-a3cc-4afd-86d9-8f35cc4a448d.png">

DawoudSheraz commented on src/users/entitlements/v2/Entitlements.jsx:
```diff
@@ -0,0 +1,298 @@
+              disabled={Boolean(!entitlement.enrollmentCourseRun)}
+              className="small"
+            >
+              Reissue
```
Lets use full words, like Reissue Entitlement

Ali-D-Akbar replied:
Hmm, I think it's pretty obvious that this action is specific to an entitlement.

DawoudSheraz replied:
Yes, it is. But take an example of the course dashboard. We see the Resume Course button there. Only Resume would have been enough too. Having explicit words is better.

DawoudSheraz reviewed (commented):
the changes look good on the surface. I have not tested it on local yet. Address the suggested changes before the final approval.

DawoudSheraz commented on src/index.jsx:
```diff
@@ -36,7 +36,7 @@ subscribe(APP_READY, () => {
           <Route exact path="/" component={SupportHomePage} />
           <Route path="/users" component={UserPage} />
-          {/* <Route path="/usersv2" component={UserPageV2} /> */}
+          <Route path="/usersv2" component={UserPageV2} />
```
comment this out.

DawoudSheraz reviewed (changes requested):
Enrollment Date and Course Run ID are two distinct headers. The course run ID column is missing.

Ali-D-Akbar commented:
> Enrollment Date and Course Run ID are two distinct headers. The course run ID column is missing.

<img width="1605" alt="Screenshot 2021-09-10 at 11 21 10 PM" src="https://user-images.githubusercontent.com/52413434/132899673-e277281a-1854-4d60-b5e4-125a39a6d49e.png">
Added course run ID and removed enrollment date for now as we don't receive this from backend.

DawoudSheraz commented on src/users/entitlements/v2/Entitlements.jsx:
```diff
@@ -0,0 +1,302 @@
+  const renderRowSubComponent = useCallback(
+    ({ row }) => (
+      <TableV2
+        // eslint-disable-next-line react/prop-types
```
is this necessary?

Ali-D-Akbar replied:
Yes, react/prop-types eslint rule says every function must have their prop-types defined. This is a callback so we may not know what each row has.

DawoudSheraz commented:
squash the commits before merging.
